src/Chainweb/Transaction.hs

```diff
@@ -42,8 +42,8 @@ import Chainweb.Utils (Codec(..))
 -- the Text that generated it, to make gossiping easier.
 --
 data PayloadWithText = PayloadWithText
-    { payloadBytes :: ByteString
-    , payloadObj :: Payload PublicMeta ParsedCode
+    { payloadBytes :: !ByteString
```

Comment:
The question on my mind is what are we doing with these payload bytes. Should this be a Short, since the only operations we do with this are decoding? (is that true?)

Reply:
Good point. I think, generally, when things are stored in memory across many major GC cycles, usage of `ShortByteString` can be a good idea.

Reply:
That's how I've been thinking about our memory problem. I wonder if the same could be said for `Text` swapping out for `Text.Short`.

Reply:
I am a bit careful with using it across the board. conversion is always a `memcpy`, so one should be a bit careful with large structure. Also, while currently, that overhead is minimal compared to other overheads, that may change in the future.

Reply:
Also my understanding is that the main benefit of short byte strings is that those are not pinned and live and the normal Haskell heap. Short text is mostly about saving a bit of memory, but not 100%.